src/security/SSHKeyManager.ts

```diff
@@ -0,0 +1,134 @@
+import * as crypto from 'crypto';
+import { getSSHConfig } from '../config';
+
+/**
+ * Secure SSH Key Manager for temporary storage of user SSH keys during approval process
+ */
+export class SSHKeyManager {
+  private static readonly ALGORITHM = 'aes-256-gcm';
+  private static readonly KEY_EXPIRY_HOURS = 24; // 24 hours max retention
+  private static readonly IV_LENGTH = 16;
+  private static readonly TAG_LENGTH = 16;
+
+  /**
+   * Get the encryption key from environment or generate a secure one
+   * @return {Buffer} The encryption key
+   */
+  private static getEncryptionKey(): Buffer {
+    const key = process.env.SSH_KEY_ENCRYPTION_KEY;
+    if (key) {
+      return Buffer.from(key, 'hex');
+    }
+
+    // For development, use a key derived from the SSH host key
+    const hostKeyPath = getSSHConfig().hostKey.privateKeyPath;
+    const fs = require('fs');
+    const hostKey = fs.readFileSync(hostKeyPath);
+
+    // Create a consistent key from the host key
+    return crypto.createHash('sha256').update(hostKey).digest();
+  }
+
+  /**
+   * Securely encrypt an SSH private key for temporary storage
+   * @param {Buffer | string} privateKey The SSH private key to encrypt
+   * @return {object} Object containing encrypted key and expiry time
+   */
+  static encryptSSHKey(privateKey: Buffer | string): {
+    encryptedKey: string;
+    expiryTime: Date;
+  } {
+    const keyBuffer = Buffer.isBuffer(privateKey) ? privateKey : Buffer.from(privateKey);
+    const encryptionKey = this.getEncryptionKey();
```

Comment:
`getEncryptionKey` says it is for development use (if the env variable isn't set), but I don't see a code path calling `generateEncryptionKey` which says it is for production. I'd also consider renaming the methods to be more clear that one is development and production, rather than get vs generate which isn't that clear...